src/xo-app/logs/index.js

```diff
@@ -85,23 +85,28 @@ const LOG_COLUMNS = [
   },
   {
     name: _('jobId'),
-    itemRenderer: log => log.jobId
+    itemRenderer: log => log.jobId,
+    sortCriteria: log => log.jobId
   },
   {
     name: _('job'),
-    itemRenderer: log => jobKeyToLabel[log.key]
+    itemRenderer: log => jobKeyToLabel[log.key],
+    sortCriteria: log => jobKeyToLabel[log.key]
```

Comment:
Ah yes, thank you, I always get caught... don't be surprised, I'll sort with a "untranslated" log.key, we recently agreed with @julien-f, on a similar case, that the item sorting should not vary with the language.